File: R ML/7_statische Grundlagen/Mittelwert_Median.R
# Damit setzen wir das Working Directory auf den Ordner dieser Datei
if (!is.null(parent.frame(2)$ofile)) {
  this.dir <- dirname(parent.frame(2)$ofile)
  setwd(this.dir)
}

# Benoetigte Libraries laden
library(ggplot2)
library(data.table)

diamonds <- fread("diamonds.csv")

print(mean(diamonds$price)) #Mittelwert
print(median(diamonds$price)) #Median

print(summary(diamonds)) #komplette Übersicht mit min, max, median, Mittelwert, ...

File: Python ML/7_statische_Grundlagen/Mittelwert_Median.py
#Mittelwert : 5 + 8 +8 +9 +5 = 35 / 5 = 7
#Median : 1 2 4 5 -> 3

import numpy as np
import pandas as pd

incomes = [10000, 30000, 5000000]
print(np.median(incomes))
print(np.mean(incomes))

df = pd.read_csv("diamonds.csv")

print(df["carat"].median())
print(df["carat"].mean())

print(df.mean()) #Mittelwert aller Spalten